main.go

```diff
@@ -202,3 +215,37 @@ func wrapAuthHandler(handler http.HandlerFunc, htpasswdFile string, host string)
 	authenticator := auth.NewBasicAuthenticator(host, secrets)
 	return auth.JustCheck(authenticator, handler)
 }
+
+func writePidFile(c *config.Config, accessLogger cache.Logger) error {
+	// create a "pid" directory under cache directory and clean up inactive pid file
+	pidPath := filepath.Join(c.Dir, "pid")
+	err := os.MkdirAll(pidPath, os.FileMode(0744))
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	err = filepath.Walk(pidPath, func(name string, fileInfo os.FileInfo, err error) error {
+		if !fileInfo.IsDir() {
+			pid, _ := strconv.Atoi(fileInfo.Name())
+			bazelRemoteProcess, err := os.FindProcess(pid)
+			if err != nil {
+				accessLogger.Printf("Error to find pid: %d", bazelRemoteProcess)
+			} else {
+				err = bazelRemoteProcess.Signal(syscall.Signal(0))
+				if err != nil && strings.Contains(err.Error(), "process already finished") {
+					accessLogger.Printf("Removing pid file: %s", fileInfo.Name())
+					err = os.Remove(filepath.Join(c.Dir, "pid", fileInfo.Name()))
+				} else if c.KillOldPid {
+					accessLogger.Printf("Killing existing bazel remote process: %d", pid)
+					err = bazelRemoteProcess.Signal(syscall.Signal(syscall.SIGKILL))
+				}
```

Comment:
If a daemon is already running and we don't want to kill it, this process should exit immediately. It seems the current code doesn't support this behavior. It just continue to check the next pid file instead.

Reply:
This will be in a seperate PR, and will be supported through a flag like:
--kill_old_pid

Reply:
Why that will be in another PR?  Checking the pid file, if one is running, exit yourself, that is the whole process.

--kill_pod_pid is about to kill other process, different thing.

Reply:
There might be use cases where we may want to keep the original process running and spawn a new process with a different port

Reply:
We need to clarify the design requirement, and the give solid use cases.

Even if that is a supported use case, it is still the most common use case that that the daemon terminate itself if there is another process of the same version already running.